src/jobs/recommendation.service.ts

```
@@ -0,0 +1,625 @@
+      relations: ['job'],
+    });
+  }
+
+  private async getUserSavedJobs(userId: string): Promise<SavedJob[]> {
+    return this.savedJobRepository.find({
+      where: { user: { id: userId } },
+      relations: ['job'],
+    });
+  }
+
+  private async getUserViewHistory(userId: string): Promise<Job[]> {
```

ESLint

[error] 'userId' is defined but never used.

(@typescript-eslint/no-unused-vars)